addons/onboarding/models/onboarding_step.py

```diff
@@ -0,0 +1,84 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, fields, models
+
+
+class OnboardingStep(models.Model):
+    _name = 'onboarding.onboarding.step'
+    _description = 'Onboarding Step'
+
+    name = fields.Char('Name of the onboarding step', translate=True)
+    onboarding_id = fields.Many2one(
+        'onboarding.onboarding', string='Onboarding', required=True, ondelete='cascade')
+
+    is_per_company = fields.Boolean(
+        related='onboarding_id.is_per_company', string='Should be done per company?')
+
+    panel_step_title = fields.Char('Title', translate=True)
+    panel_step_description = fields.Char('Description', translate=True)
+    panel_step_button_text = fields.Char('Button text', default="Let's do it", translate=True)
+    panel_step_done_icon = fields.Char('Font Awesome Icon when completed', default='fa-star')
+    panel_step_done_text = fields.Char(
+        'Text to show when step is completed', default='Step Completed! - Click to review', translate=True)
+    panel_step_open_action_name = fields.Char(
+        string='Opening action', required=True,
+        help='Name of the onboarding step model action to execute when opening the step, '
+             'e.g. action_open_onboarding_1_step_1')
+
+    current_progress_id = fields.One2many(
```

Comment:
Should probably be a related on onboarding, as you compute several times the same thing.